app/Notifications/Channels/ApnsChannel.php

```
@@ -0,0 +1,143 @@
+            return;
+        }
+
+        $existing = $message->custom['spark'] ?? [];
+
+        $message->custom('spark', array_merge($envelope, $existing));
+    }
+
+    /**
+     * Dispatch a silent content-available push so the client can sync.
+     */
+    protected function sendSilentCompanion(Client $client, mixed $notifiable, Notification $notification, array $tokens): void
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$notifiable'. (undefined)

(UnusedFormalParameter)